src/V3Param.cpp

```diff
@@ -965,11 +1102,15 @@ class ParamVisitor final : public VNVisitor {
     }
 
     virtual void visit(AstClassRefDType* nodep) override {
-        if (nodep->paramsp()) {
-            nodep->paramsp()->v3warn(E_UNSUPPORTED, "Unsupported: parameterized classes");
-            pushDeletep(nodep->paramsp()->unlinkFrBackWithNext());
-        }
-        iterateChildren(nodep);
+        string* const genHierNamep = new string(m_generateHierName);
```

Comment:
Not sure about this. Could we put this string in a struct together with the class ref in `m_classRefps`? I don't think there's any advantage to using `user*` here.

Reply:
This is basically doing the same thing that `visit(AstCell*)` did, and I am not sure why it happens exactly.
The two lists could probably get merged together.